compiler/rustc_middle/src/query/mod.rs

```diff
@@ -222,6 +222,7 @@ rustc_queries! {
 
     /// Fetch the THIR for a given body. If typeck for that body failed, returns an empty `Thir`.
     query thir_body(key: ty::WithOptConstParam<LocalDefId>) -> (&'tcx Steal<thir::Thir<'tcx>>, thir::ExprId) {
+        no_hash
```

Comment:
Since it's non-obvious why this is `no_hash`, would you mind adding a comment explaining the reasoning?

Reply:
r=me with this comment :)

Reply:
Would something like this make sense?
```suggestion
        // Perf tests revealed that hashing THIR is inefficient (see #85729).
        no_hash
```